File: src/test/java/com/bonepl/chromaleague/razer/effects/OneKeyEffectTest.java
package com.bonepl.chromaleague.razer.effects;

import com.bonepl.chromaleague.razer.RazerSDKClient;
import com.bonepl.chromaleague.razer.effects.keyboard.OneKeyEffect;
import com.bonepl.chromaleague.razer.sdk.RzKey;
import org.junit.jupiter.api.Test;

class OneKeyEffectTest {

    @Test
    void testOneKeyEffect() throws InterruptedException {
        try (RazerSDKClient razerSDKClient = new RazerSDKClient()) {
            for (RzKey rzKey : RzKey.values()) {
                razerSDKClient.createKeyboardEffect(new OneKeyEffect(rzKey, Color.RED));
                Thread.sleep(50);
            }
        }
    }
}

File: test/razer-sdk/effects/OneKeyEffect.test.ts
import {describe, test} from 'vitest';
import {ChromaNativeSDK} from '../../../src/razer-sdk/ChromaNativeSDK.js';
import {SimpleFrame} from '../../../src/razer-sdk/animation/SimpleFrame.js';
import {StaticColor} from '../../../src/razer-sdk/color/StaticColor.js';
import {getRzKeyValues} from '../../../src/razer-sdk/sdk/RzKey.js';

/**
 * Integration test for individual key lighting on actual hardware.
 * Lights up each key one by one across the entire keyboard.
 */
describe('OneKeyEffect Integration', () => {
    test('testOneKeyEffect', async () => {
        const sdk = new ChromaNativeSDK();
        await sdk.connect();

        try {
            for (const rzKey of getRzKeyValues()) {
                await sdk.createKeyboardEffect(new SimpleFrame(rzKey, StaticColor.RED));
                await new Promise(resolve => setTimeout(resolve, 50));
            }
        } finally {
            await sdk.close();
        }
    }, 60000);
});
